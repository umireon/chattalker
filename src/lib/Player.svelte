<script lang="ts">
  import 'three-dots/dist/three-dots.min.css'

  export let playerIsLoading: boolean
  export let playerLanguage: string
  export let playerSrc: string
  export let playerText: string
</script>

<main>
  <p><audio src={playerSrc} autoplay></audio></p>
  <div style="display: flex; align-items: flex-start;">
    <p style="margin: 0px;">
      <span id="language">{playerLanguage}</span>
      <span id="text">{playerText}</span>
    </p>
    <div class="dot-bricks" style="visibility: {playerIsLoading ? 'visible' : 'hidden'};"></div>
  </div>
</main>

<style>
  .dot-bricks {
    margin: 10px;
  }
</style>