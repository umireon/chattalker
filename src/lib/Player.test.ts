import { expect, test } from "vitest";

import Player from "./Player.svelte";
import { render } from "@testing-library/svelte";

test("Player snapshot", () => {
  const props = {
    playerIsLoading: false,
    playerLanguage: "",
    playerSrc: "",
    playerText: "",
  };
  const component = render(Player, { props });
  expect(component.container).toMatchSnapshot();
});
