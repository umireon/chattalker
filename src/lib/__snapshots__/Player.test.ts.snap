// Vitest Snapshot v1

exports[`Player > matches snapshot 1`] = `
<body>
  <div>
    <main>
      <p>
        <audio
          autoplay=""
          src=""
          type="audio/mpeg"
        />
      </p>
       
      <div
        style="display: flex; align-items: flex-start;"
      >
        <p
          style="margin: 0px;"
        >
          <span
            id="language"
          >
            
          </span>
           
          <span
            id="text"
          >
            
          </span>
        </p>
         
        <div
          class="dot-bricks svelte-vdl2sx"
          style="visibility: hidden;"
        />
      </div>
    </main>
  </div>
</body>
`;
